lxd/db/cluster/cluster_groups.go

```diff
@@ -51,3 +57,39 @@ func (c *ClusterGroup) ToAPI() (*api.ClusterGroup, error) {
 
 	return &result, nil
 }
+
+// GetClusterGroupUsedBy collates references to the cluster group with the given name.
+// This currently only returns the URLs of projects whose `restricted.cluster.groups` configuration
+// contains the cluster group.
+func GetClusterGroupUsedBy(ctx context.Context, tx *sql.Tx, groupName string) ([]string, error) {
+	q := `
+SELECT projects.name, projects_config.value FROM projects 
+JOIN projects_config ON projects.id = projects_config.project_id 
+WHERE projects_config.key = 'restricted.cluster.groups'`
+
+	stmt, err := tx.Prepare(q)
+	if err != nil {
+		return nil, err
+	}
+
+	var projectURLs []string
+	err = query.SelectObjects(ctx, stmt, func(scan func(dest ...any) error) error {
+		var projectName string
+		var configValue string
+		err := scan(&projectName, &configValue)
+		if err != nil {
+			return err
+		}
+
+		if shared.ValueInSlice(groupName, shared.SplitNTrimSpace(configValue, ",", -1, false)) {
+			projectURLs = append(projectURLs, entity.ProjectURL(projectName).String())
```

Comment:
I'm interested why you convert the api.URL to a string here rather than return a slice of api.URLs themselves, I feel that would be clearer as to the contents of the slice rather than a string?

Reply:
I was just following the convention of other `UsedBy` methods but happy to change.